### ISSUE TITLE
Modeling

### ISSUE THREAD

**Issue by tutorialcreation** [tutorialcreation/abtest-mlops#6]
perform the following functions

Split data by browser and platform_os, and version each split as a new version of the data in dvc.
For each version of the data do the following
Split the data into 70% training, 20% validation, and 10% test sets. 
Based on the reading material provided, apply machine learning to the training data. Train a machine learning model using 5-fold cross validation using the following 3 different algorithms:
Logistic Regression 
Decision Trees
XGBoost
RandomForest
Define the appropriate loss function  for the model using the validation data. 
Compute feature importance - what’s driving the model? Which parameters are important predictors for the different ML models? What contributes to the goal of gaining more “Yes” results?
Which data features are relevant to predicting the target variable?
For each of the ML algorithms above, find the best model by tuning their hyperparameters and each time adding the tried models in MLFlow.
Prepare a Dockerfile for your project so that your model can be deployed in a docker container.
